Looms.PoS/Looms.PoS.Application/Features/Product/Commands/CreateProduct/CreateProductCommandHandler.cs

```diff
@@ -13,25 +14,35 @@ public class CreateProductCommandHandler : IRequestHandler<CreateProductCommand,
     private readonly IProductsRepository _productsRepository;
     private readonly IHttpContentResolver _httpContentResolver;
     private readonly IProductModelsResolver _modelsResolver;
+    private readonly ITaxesRepository _taxesRepository;
 
     public CreateProductCommandHandler(
         IProductsRepository productsRepository,
         IHttpContentResolver httpContentResolver,
-        IProductModelsResolver modelsResolver)
+        IProductModelsResolver modelsResolver,
+        ITaxesRepository taxesRepository)
     {
         _productsRepository = productsRepository;
         _httpContentResolver = httpContentResolver;
         _modelsResolver = modelsResolver;
+        _taxesRepository = taxesRepository;
     }
 
     public async Task<IActionResult> Handle(CreateProductCommand command, CancellationToken cancellationToken)
     {
         var productRequest = await _httpContentResolver.GetPayloadAsync<CreateProductRequest>(command.Request);
 
-        var createdProductDao = _modelsResolver.GetDaoFromRequest(productRequest);
+        TaxDao tax;
 
-        var response = _modelsResolver.GetResponseFromDao(createdProductDao);
+        if (productRequest.TaxId is null)
+        {
+           tax = await _taxesRepository.GetByTaxCategoryAsync(TaxCategory.Product);
+        } else {
+           tax = await _taxesRepository.GetAsync(Guid.Parse(productRequest.TaxId));
+        }
+        
+        var productDao = _modelsResolver.GetDaoFromRequest(productRequest, tax);
 
-        return new CreatedAtRouteResult($"/products/{createdProductDao.Id}", response);
+        return new CreatedAtRouteResult($"/products{productDao.Id}", productDao);
```

Comment:
```suggestion
        return new CreatedAtRouteResult($"/products/{productDao.Id}", productDao);
```

Reply:
changed

Reply:
Product is never saved in CreateProductCommandHandler